test/test_common.py

```diff
@@ -88,6 +90,40 @@ def test_json_serialization(self, tmpdir):
 
         assert ob2 == ob2_
 
+    def test_persisted_progress(self):
+        test_progress_file = "test_progress.json"
+
+        test_data = {
+            "progress": {
+                "0af668268d0fe14c6e269760b08d80a634c421b8381df25f31fbed5e8a8c8d8b": {
+                    "current_progress": 100,
+                    "last_updated_time": 1625072042.123456
+                }
+            }
+        }
+
+        with open(test_progress_file, "w") as file:
+            json.dump(test_data, file)
+
+        with patch("pghoard.common.PROGRESS_FILE", test_progress_file):
+            persisted_progress = PersistedProgress.read(metrics=metrics.Metrics(statsd={}))
+            assert "0af668268d0fe14c6e269760b08d80a634c421b8381df25f31fbed5e8a8c8d8b" in persisted_progress.progress
+            assert persisted_progress.progress["0af668268d0fe14c6e269760b08d80a634c421b8381df25f31fbed5e8a8c8d8b"
+                                               ].current_progress == 100
+            assert persisted_progress.progress["0af668268d0fe14c6e269760b08d80a634c421b8381df25f31fbed5e8a8c8d8b"
+                                               ].last_updated_time == 1625072042.123456
+
+            new_progress = 200
+            progress_info = persisted_progress.get("0af668268d0fe14c6e269760b08d80a634c421b8381df25f31fbed5e8a8c8d8b")
+            progress_info.update(new_progress)
+            persisted_progress.write(metrics=metrics.Metrics(statsd={}))
+
+            updated_progress = PersistedProgress.read(metrics=metrics.Metrics(statsd={}))
+            assert updated_progress.progress["0af668268d0fe14c6e269760b08d80a634c421b8381df25f31fbed5e8a8c8d8b"
```

Comment:
you should check that the last_update time has changed as well

Reply:
added